src/rgw/rgw_sal_motr.cc

```diff
@@ -1500,6 +1500,14 @@ MotrObject::MotrDeleteOp::MotrDeleteOp(MotrObject *_source, RGWObjectCtx *_rctx)
 // 2. Delete an object when its versioning is turned on.
 int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional_yield y)
 {
+  if (source->have_instance()) {
+    rgw_obj_key& key = source->get_key();
+    const std::string& instance = key.get_instance();
+    if (instance == "null") {
+      key.set_instance("");
```

Comment:
Why not `key.instance.clear()` ?

Also, style: if there is only one line under `if` - there is no need for braces:

```
if (instance == "null")
  key.instance.clear()
```

Reply:
Done the change.
I thought that using getters and setters would be better than accessing the internal member of structure.